apps/crn-server/src/migrations/1665394546-move-type-tutorial-news-to-entity-tutorials.ts

```diff
@@ -0,0 +1,64 @@
+/* istanbul ignore file */
+import {
+  RestNews,
+  Results,
+  SquidexRest,
+  RestTutorials,
+} from '@asap-hub/squidex';
+import { appName, baseUrl } from '../config';
+import { Migration } from '../handlers/webhooks/webhook-run-migrations';
+import { getAuthToken } from '../utils/auth';
+
+export default class MoveTypeTutorialNewsToEntityTutorials extends Migration {
+  up = async (): Promise<void> => {
+    const squidexNewsClient = new SquidexRest<RestNews>(
+      getAuthToken,
+      'news-and-events',
+      { appName, baseUrl },
+      {
+        unpublished: true,
+      },
+    );
+
+    const squidexTutorialsClient = new SquidexRest<RestTutorials>(
+      getAuthToken,
+      'tutorials',
+      { appName, baseUrl },
+      {
+        unpublished: true,
+      },
+    );
+
+    let pointer = 0;
+    let result: Results<RestNews>;
+
+    do {
+      result = await squidexNewsClient.fetch({
+        $top: 10,
+        $skip: pointer,
+        $orderby: 'created asc',
+      });
+
+      for (const news of result.items) {
+        if (
+          news.data.type !== ('Tutorial' as unknown as RestNews['data']['type'])
+        ) {
+          continue;
+        }
+
+        const { type, frequency, ...tutorialData } = news.data;
+
+        try {
+          await squidexTutorialsClient.create(tutorialData);
+          await squidexNewsClient.delete(news.id);
+        } catch (err) {
+          console.log('error: ', err);
+        }
+      }
+
+      pointer += 10;
+    } while (pointer < result.total);
+  };
+
+  down = async (): Promise<void> => {};
```

Comment:
do we want a rollback script in this case? also, what happens if the script fails in the middle? do we want to save the migrated news ids somewhere?

Reply:
the migrated news will be whatever the record where moved into tutorials, to be fair the rollback is there but i dont think we ever used it, but if you wanted to feel safe i'd say just move all records from tutorials back to news and delete them from tutorials. Or leave it empty, whatever, but then make sure to have a backup before the migration. In fact - have it anyway